#! /bin/sh /usr/share/dpatch/dpatch-run
## 26_users_home_dir.dpatch by  <james.westby@canonical.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Set the home directory to that of the user if it doesn't already exist

@DPATCH@
diff -urNad gnome-system-tools-2.22.0~/src/users/user-settings.c gnome-system-tools-2.22.0/src/users/user-settings.c
--- gnome-system-tools-2.22.0~/src/users/user-settings.c	2008-03-31 16:32:35.000000000 +0100
+++ gnome-system-tools-2.22.0/src/users/user-settings.c	2008-03-31 16:32:36.000000000 +0100
@@ -347,9 +347,6 @@
 		widget = gst_dialog_get_widget (tool->main_dialog, "user_settings_shell");
 		set_entry_text (GTK_BIN (widget)->child, oobs_user_get_shell (user));
 
-		widget = gst_dialog_get_widget (tool->main_dialog, "user_settings_home");
-		set_entry_text (widget, oobs_user_get_home_directory (user));
-
 		widget = gst_dialog_get_widget (tool->main_dialog, "user_settings_uid");
 		gtk_spin_button_set_value (GTK_SPIN_BUTTON (widget), oobs_user_get_uid (user));
 		gtk_widget_set_sensitive (GTK_SPIN_BUTTON (widget), FALSE);
@@ -394,6 +391,11 @@
 	if (!login)
 		table_set_default_profile (GST_USERS_TOOL (tool));
 
+	if (user) {
+		widget = gst_dialog_get_widget (tool->main_dialog, "user_settings_home");
+		set_entry_text (widget, oobs_user_get_home_directory (user));
+	}
+
 	return dialog;
 }
 
